helpers/helpers.py

```diff
@@ -20,8 +20,160 @@
+        logger.error(f"Error modifying space settings: {e}")
+        raise
+
+
+def create_snapshot_proposal(proposal_data, title):
+    """
+    Create a Snapshot proposal using the proposal data.
+    Called from tasks/tasks.py.
+
+    Parameters:
+    proposal_data (Dict): The proposal data to create the proposal with.
+    title (str): The title of the proposal.
+
+    Raises:
+    subprocess.CalledProcessError: If an error occurs while creating the proposal
+    """
+    proposal_command = [
+        "node",
+        "./snapshot/wrapper.js",
+        title,
+        proposal_data["draft"]["abstract"],
+        proposal_data["draft"]["background"],
+        proposal_data["draft"]["additional"],
+        "Adopt",
+        "Reassess",
+        "Abstain",
+    ]
+
+    try:
+        subprocess.run(proposal_command, check=True)
+        logger.info("Snapshot proposal created successfully.")
+    except subprocess.CalledProcessError as e:
+        logger.error(f"Error creating snapshot proposal: {e}")
+        raise
+
+
+def fetch_XP_total_supply() -> int:
+    """
+    Fetch the total supply of tokens across all specified addresses and convert it to ether.
+
+    Returns:
+    int: The total supply of all tokens converted to ether.
+    """
+    # Fetch RPC URLs from environment variables
+    primary_rpc = os.getenv("PRIMARY_RPC")
+    secondary_rpc = os.getenv("SECONDARY_RPC")
+
+    if not primary_rpc or not secondary_rpc:
+        logger.error("RPC URLs not set in environment variables.")
+        return None
+
+    web3 = Web3(Web3.HTTPProvider(primary_rpc))
+
+    # Check if we are connected to the node
+    if not web3.is_connected():
+        logger.error("Failed to connect to PRIMARY_RPC, let's try the SECONDARY_RPC")
+        web3 = Web3(Web3.HTTPProvider(secondary_rpc))
+        if not web3.is_connected():
+            logger.error("Failed to connect to SECONDARY_RPC, shutting down")
+            exit(1)
+
+    token_abi = [
+        {
+            "constant": True,
+            "inputs": [],
+            "name": "totalSupply",
+            "outputs": [{"name": "", "type": "uint256"}],
+            "type": "function",
+        }
+    ]
+
+    token_addresses = [
+        "0x206d247F61cb82B9711318381cDb7Bc5039d2A2c",
+        "0x4cd06ada7d8564830018000d784c69bd542b1e6a",
+        "0x57d3a929fdc4faf1b35e7092d9dee7af097afb6a",
+    ]
+
+    # Initialize total supply counter
+    total_supply_sum = 0
+
+    # Loop over each token address and fetch the total supply, converting it to ether
+    for address in token_addresses:
+        try:
+            # Convert to checksum address
+            checksum_address = Web3.to_checksum_address(address)
+            token_contract = web3.eth.contract(address=checksum_address, abi=token_abi)
+            total_supply = token_contract.functions.totalSupply().call()
+            if total_supply is not None:
+                total_supply_in_ether = web3.from_wei(total_supply, "ether")
```

Comment:
imo this function should not be converting the value to "ether". the expected output of a total supply function is the total supply in it's base unit.

Reply:
> imo this function should not be converting the value to "ether". the expected output of a total supply function is the total supply in it's base unit.

Converting to ether works better when setting the quorum value to update the Snapshot space. For example if we use 'wei' the number returned is: 25049774290000000000000000 which would then be set as the quorum value. This far exceeds the actual supply of tokens.  (https://github.com/snapshot-labs/snapshot.js/blob/master/src/schemas/space.json)

```
            "quorum": {
              "type": "number",
              "minimum": 0
            },
```

Using ether returns 25049774 which is indeed 25% of the supply at current values.

Reply:
Yea I understand. I'm suggesting this function should return the actual value, and then you should convert to ether in "fetch_xp_quorum"

Reply:
> Yea I understand. I'm suggesting this function should return the actual value, and then you should convert to ether in "fetch_xp_quorum"

Suspected that may have been your response, makes sense 👍

Reply:
> Yea I understand. I'm suggesting this function should return the actual value, and then you should convert to ether in "fetch_xp_quorum"

Ether conversion is now done within fetch_xp_quorum